src/ring_theory/polynomial/cyclotomic.lean

```diff
@@ -0,0 +1,452 @@
+comes from a polynomial with integer coefficients.
+* `deg_of_cyclotomic` : The degree of `cyclotomic n` is `totient n`.
+* `X_pow_sub_one_eq_prod_cycl` : `X ^ n - 1 = ∏ (cyclotomic i)`, where `i` divides `n`.
+
+## Implementation details
+
+Our definition of `cyclotomic' n R` makes sense in any integral domain `R`, but the interesting
+results hold if there is a primitive `n`th root of unity in `R`. In particular, our definition is
+not the standard one unless there is a primitive `n`th root of unity in `R`. For example,
+`cyclotomic' 3 ℤ = 1`, since there are no primitive cube roots of unity in `ℤ`. The main example is
+`R = ℂ`, we decided to work in general since the difficulties are essentially the same.
+To get the standard cyclotomic polynomials, we use `int_coeff_of_cycl`, with `R = ℂ`, to get a
+polynomial with integer coefficients and then we map it to `polynomial R`, for any ring `R`.
+
+-/
+
+open_locale classical big_operators
+noncomputable theory
+
+universe u
+
+namespace polynomial
+
+section cyclotomic'
+
+section integral_domain
+
+variables {R : Type*} [integral_domain R]
+
+/-- The modified `n`-th cyclotomic polynomial with coefficients in `R`, it is the usual cyclotomic
+polynomial if there is a primitive `n`-th root of unity in `R`. -/
+def cyclotomic' (n : ℕ) (R : Type*) [integral_domain R] : polynomial R :=
+∏ μ in primitive_roots n R, (X - C μ)
+
+/-- The zeroth modified cyclotomic polyomial is `1`. -/
+@[simp] lemma cyclotomic'_zero (R : Type*) [integral_domain R] : cyclotomic' 0 R = 1 :=
+by simp only [cyclotomic', finset.prod_empty, is_primitive_root.primitive_roots_zero]
+
+/-- The first modified cyclotomic polyomial is `X - 1`. -/
+@[simp] lemma cyclotomic'_one (R : Type*) [integral_domain R] : cyclotomic' 1 R = X - 1 :=
+begin
+  simp only [cyclotomic', finset.prod_singleton, ring_hom.map_one,
+  is_primitive_root.primitive_roots_one]
+end
+
+/-- The second modified cyclotomic polyomial is `X + 1` if the characteristic of `R` is not `2`. -/
+@[simp] lemma cyclotomic'_two (R : Type*) [integral_domain R] (p : ℕ) [char_p R p] (hp : p ≠ 2) :
+  cyclotomic' 2 R = X + 1 :=
+begin
+  rw [cyclotomic'],
+  have prim_root_two : primitive_roots 2 R = {(-1 : R)},
+  { apply finset.eq_singleton_iff_unique_mem.2,
+    split,
+    { simp only [is_primitive_root.neg_one p hp, nat.succ_pos', mem_primitive_roots] },
+    { intros x hx,
+      rw [mem_primitive_roots zero_lt_two] at hx,
+      exact is_primitive_root.eq_neg_one_of_two_right hx } },
+  simp only [prim_root_two, finset.prod_singleton, ring_hom.map_neg, ring_hom.map_one,
+  sub_neg_eq_add]
+end
+
+/-- `cyclotomic' n R` is monic. -/
+lemma cyclotomic'.monic (n : ℕ) (R : Type*) [integral_domain R] : (cyclotomic' n R).monic :=
+monic_prod_of_monic _ _ $ λ z hz, monic_X_sub_C _
+
+/-- `cyclotomic' n R` is different from `0`. -/
+lemma cyclotomic'_ne_zero (n : ℕ) (R : Type*) [integral_domain R] : cyclotomic' n R ≠ 0 :=
+  monic.ne_zero (cyclotomic'.monic n R)
```

Comment:
I think you can write this
```suggestion
(cyclotomic'.monic n R).ne_zero
```
(also mathlib style has no indent here, whether that works or not)